### ISSUE TITLE
predict方法直接支持处理n张opencv格式的图片组成的shape为(n,H,W,C)的numpy数组

### ISSUE THREAD

**Issue by johnfans** [ultralytics/ultralytics#23091]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

对于一些从环形缓冲区的数据，直接使用numpy会处理成shape为(n,H,W,C)的数组。按照当前版本的支持，需要先使用list()函数处理成n个元素的shape为(H,W,C)的numpy数组的原生python列表，再使用predict方法进行推理，非常的浪费性能。希望predict方法能直接支持输入shape为(n,W,H,C)的numpy数组。

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @johnfans, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the clear feature request around `predict()` accepting a batched NumPy array shaped like `(n, H, W, C)` directly (instead of requiring a Python `list` of `(H, W, C)` arrays). ✅ This response is automated 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us evaluate the request quickly, please share a small snippet that demonstrates:
-Your current input pipeline (how you build the `(n, H, W, C)` array from the ring buffer)
-The exact `predict()` call you’re using
-Your environment details (OS, Python, `ultralytics` version)
-A simple timing comparison showing the overhead you’re seeing when converting via `list()` (even rough numbers help) ⏱️

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by johnfans** [comment]
# 相关代码

## 读取buffer

```python
imgs_tmp = np.ndarray((remain_num,960,960,3),dtype=np.uint8,buffer=self.shm.buf,offset=self.ptr*(960*960*3))
```
其中remain_num为读取的图片个数，shm为环形缓冲区的共享内存，ptr为“指针”

##  现在的推理方法

```python
results = model.predict(list(imgs_tmp),stream=False,conf=0.3,iou=0.5,imgsz=960, task='detect')
```

## 期望的推理方法
```python
results = model.predict(imgs_tmp,stream=False,conf=0.3,iou=0.5,imgsz=960, task='detect')
```

**Comment by Y-T-G** [comment]
There wouldn't be any benefit, because internally, the numpy array will still need to be split into a list because each image needs to be preprocessed separately since preprocessing is done using OpenCV which doesn't support batched transformation.

If you want to speed it up, you can perform preprocessing manually and then use the batched tensor for inference. For tensor input, preprocessing is skipped.

**Comment by johnfans** [comment]
> There wouldn't be any benefit, because internally, the numpy array will still need to be split into a list because each image needs to be preprocessed separately since preprocessing is done using OpenCV which doesn't support batched transformation.
> 
> If you want to speed it up, you can perform preprocessing manually and then use the batched tensor for inference. For tensor input, preprocessing is skipped.

Does the predict method support direct input of tensors? What is the best practice for converting images into tensors in batches?（opencv format）

**Comment by glenn-jocher** [comment]
Yes—`model.predict()` accepts a batched `torch.Tensor` input (BCHW), and when you pass a tensor Ultralytics skips the usual OpenCV/numpy preprocessing, so you should provide RGB + normalized `float` (0–1) at the final size you want (i.e., `imgsz` won’t letterbox/resize tensor inputs). See the supported inference sources in the [Predict mode docs](https://docs.ultralytics.com/modes/predict/).

```python
import torch
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

# imgs_tmp: (N, H, W, 3) uint8, OpenCV BGR
imgs_rgb = imgs_tmp[..., ::-1].copy()  # BGR->RGB (copy keeps it contiguous)

# (N, H, W, C) -> (N, C, H, W), normalize to 0..1 float
im = torch.from_numpy(imgs_rgb).permute(0, 3, 1, 2).contiguous().float() / 255.0

results = model.predict(im, conf=0.3, iou=0.5, stream=False)
```
